### PR TITLE
use contextual logging

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Use contextual logging as per Cluster API/Kubernetes logging guidelines

**Which issue this PR fixes**: 
fixes #

**Special notes for your reviewer**:

**Release notes**:
```release-note
Controllers are now using contextual logging
```

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -4,132 +4,194 @@ go 1.17
 
 require (
 	github.com/Masterminds/semver v1.5.0
-	github.com/golangci/golangci-lint v1.32.0
+	github.com/golangci/golangci-lint v1.48.0
```
Comment (fabriziopandini) [inline hack/tools/go.mod:7]: bumped otherwise lint was failing on M!

File: hack/tools/go.mod [hunk 2]
```diff
@@ -4,132 +4,194 @@ go 1.17
...
 	github.com/google/go-containerregistry v0.1.4
 	github.com/spf13/pflag v1.0.5
+	k8s.io/code-generator v0.25.0
```
Comment (fabriziopandini) [inline hack/tools/go.mod:10]: Added to get make generate-manifests/make verify to run

**Review by elmiko — COMMENTED** [review]
/lgtm happy to relabel after rebase

**Review by elmiko — COMMENTED** [review]
/lgtm

**Pull request merged**